lld/ELF/ICF.cpp

```diff
@@ -374,6 +376,8 @@ template <class ELFT>
 bool ICF<ELFT>::equalsVariable(const InputSection *a, const InputSection *b) {
   const RelsOrRelas<ELFT> ra = a->template relsOrRelas<ELFT>();
   const RelsOrRelas<ELFT> rb = b->template relsOrRelas<ELFT>();
+  if (ra.areRelocsCrel())
```

Comment:
Sam's question here.

Reply:
Not needed. The double rel check is for https://github.com/llvm/llvm-project/commit/686cff17cc310884e48ae963bf7507f96950cc90 .

An early areRelocsCrel() issue would not run into that issue.